### DISCUSSION TITLE
Add Remote Room Sensor

### DISCUSSION THREAD

**Discussion by wriede** [PeteRager/lennoxs30#286] [General]
I recently got an S40 to replace one of my S30s whose hub had failed and it was easily recognized when I added it to this integration. But I today got a Remote Room Sensor which I got added to the S40 but I don't see it in Home Assistant. You support them so I must be missing something. How do I get the Remote to show up in HA?

**Comment by PeteRager** [comment]
It should show up automatically and there should be a device created.  What do you see there?

Any warnings in the log?

Did you create a new integration for the S40 or replace it by using the same IP address.  If the later the integration will need to be reloaded / HA restarted.  Thought that upgrade path is not well tested it should work.  If that does not work, try deleting that integration entry and readding

**Comment by wriede** [comment]
When the S40 was being installed I deleted the old S30 and later once the new S40 was up and running and coaxed to have the same IP address as the S30 had before I added the S40 as a new device.

When I added the Remote Room Sensor to the S40 nothing about it in HA changed. This what I have for its devices:

![Screenshot from 2024-01-12 07-39-20](https://github.com/PeteRager/lennoxs30/assets/107369389/0f5bfa76-44cc-42e0-8b52-0967749012eb)

I the logbook I see nothing other than some connectivity issues:

![Screenshot from 2024-01-12 07-38-24](https://github.com/PeteRager/lennoxs30/assets/107369389/57112130-adce-414b-b9d5-475a0adba924)

Is deleting the S40 and re-adding my only recourse or is there something else you want to see or trick to try?

Thanks, Willem Riede.

**Comment by wriede** [comment]
What-do-you-know :-)  I just updated something in HA that required that I restart HA and bingo: there is my Room Sensor:

![Screenshot from 2024-01-12 07-57-27](https://github.com/PeteRager/lennoxs30/assets/107369389/f2a3d06a-0d8e-4084-80fa-f803e1bb4878)

Perhaps you can add a note to the documentation to restart the integration if the sensor doesn't show up automatically?

Thanks again.

**Comment by PeteRager** [comment]
Glad it sorted itself out.  It's only at integration reload or HA restart that the devices / sensors are reevaluated- as typically this information is static.  I will add a note to the docs.  

Keep an eye on those disconnects and the attributes in that sensor that have the error counts.  It really should stay connected most of the time.  A low-level can be typically (1x a day).  It could mean a weak WiFi connection.  Or the S40 may be rebooting, in that case the system uptime counter should be reset and a warning logged.
